src/Common/Telemetry/Migration.php

```diff
@@ -0,0 +1,320 @@
+		// Store a backup of the original option.
+		update_option( $this->fs_plugins_slug, $fs_active_plugins );
+
+		foreach ( $this->our_plugins as $plugin ) {
+			$plugin .= '/common/vendor/freemius';
+
+			unset( $fs_active_plugins->plugins[$plugin] );
+
+			if ( ! empty( $fs_active_plugins->newest->sdk_path ) && $fs_active_plugins->newest->sdk_path === $plugin ) {
+				unset( $fs_active_plugins->newest );
+			}
+		}
+
+		// Update the option in the database with our edits.
+		update_option( 'fs_active_plugins', $fs_active_plugins );
+	}
+
+	/**
+	 * Removes our plugins from the fs_accounts option.
+	 *
+	 * @since 5.0.17
+	 *
+	 * @param array<string,mixed> $fs_accounts
+	 */
+	private function handle_fs_accounts( $fs_accounts ): void {
+		// Store a backup of the original option.
+		update_option( $this->fs_accounts_slug, $fs_accounts );
+
+		foreach ( $this->our_plugins as $plugin ) {
+			$fs_accounts = $this->strip_plugin_from_fs_accounts( $plugin, $fs_accounts );
+		}
+
+		// Update the option in the database with our edits.
+		update_option( 'fs_accounts', $fs_accounts );
+	}
+
+	/**
+	 * Removes all freemius options from the database.
+	 * Only used if we were the only active plugin.
+	 *
+	 * @since 5.0.17
+	 */
+	private function remove_all_freemius(): void {
+		delete_option( 'fs_active_plugins' );
+		delete_option( 'fs_accounts' );
+		delete_option( 'fs_api_cache' );
+		delete_option( 'fs_debug_mode' );
+		delete_option( 'fs_gdpr' );
+	}
+
+	/**
+	 * Contains all the logic for stripping our plugins from the fs_accounts option.
+	 *
+	 * @since 5.0.17
+	 *
+	 * @param string $plugin
+	 * @param array<string,mixed> $fs_accounts
+	 *
+	 * @return array<string,mixed>
+	 */
+	private function strip_plugin_from_fs_accounts( $plugin, $fs_accounts ): array {
+		foreach( $fs_accounts[ 'id_slug_type_path_map' ] as $key => $data ) {
+			if ( $data['slug'] === $plugin ) {
+				unset( $fs_accounts[ 'id_slug_type_path_map' ][$key] );
+			}
+		}
+
+		// these use the slug as the key.
+		$straight_keys = [
+			'plugins',
+			'plugin_data',
+			'plans',
+			'admin_notices',
+			'sites'
+		];
+
+		foreach( $straight_keys as $key ) {
+			if ( isset( $fs_accounts[$key][$plugin] ) ) {
+				unset( $fs_accounts[$key][$plugin] );
+			}
+		}
+
+		// These use the path instead of the slug as the key.
+		$plugin = $plugin . '/' . $plugin . '.php';
+		if ( isset( $fs_accounts['file_slug_map'][$plugin] ) ) {
+			unset( $fs_accounts['file_slug_map'][$plugin] );
+		}
+
+		if ( isset( $fs_accounts['active_plugins']->plugins[$plugin] ) ) {
+			unset( $fs_accounts['active_plugins']->plugins[$plugin] );
+		}
+
+		// ...and return!
+		return $fs_accounts;
+	}
+
+	/**
+	 * Opts the user in to Telemetry.
+	 *
+	 * @since 5.0.17
+	 */
+	public function auto_opt_in() {
+		$Opt_In_Subscriber = Config::get_container()->get( Opt_In_Subscriber::class );
+		$Opt_In_Subscriber->opt_in( Telemetry::get_stellar_slug() );
```

Comment:
This variable is camel cased. Can we change that?
```suggestion
		$opt_in_subscriber = Config::get_container()->get( Opt_In_Subscriber::class );
		$opt_in_subscriber->opt_in( Telemetry::get_stellar_slug() );
```